### ISSUE TITLE
Error when attempting to overlay DS9 region file

### ISSUE THREAD

**Issue by epaell** [aplpy/aplpy#369]
I've just installed py34-astropy (2.0.2_0), py34-pyregion (2.0_0) and py34-aplpy (1.1.1_0+avm+ds9+rgb) and have tried to run some of my scripts that worked previously (they simply load a FITS image file and overlay a DS9 region file) but I get the following error:

Traceback (most recent call last):
  File "./make_pngs.py", line 55, in <module>
    f1.show_regions("cat.reg")
  File "<string>", line 2, in show_regions
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/aplpy/decorators.py", line 25, in _auto_refresh
    return f(*args, **kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/aplpy/regions.py", line 65, in show_regions
    PC, TC = ds9(region_file, wcs.WCS(self._header).sub([wcs.WCSSUB_CELESTIAL]), **kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/aplpy/regions.py", line 103, in ds9
    rrim = rr.as_imagecoord(header)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/pyregion/core.py", line 70, in as_imagecoord
    shape_list, comment_list = zip(*list(r))
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/pyregion/ds9_region_parser.py", line 187, in sky_to_image
    new_coords = convert_to_imagecoord(shape, header)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/pyregion/wcs_converter.py", line 81, in convert_to_imagecoord
    new_wcs = WCS(header)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/astropy/wcs/wcs.py", line 395, in __init__
    for dict_key in orig_header.keys():
AttributeError: 'WCS' object has no attribute 'keys'

I thought perhaps it was a Python 3.4 thing but I noticed the same occurs with Python 2.7.

**Comment by epaell** [comment]
In case it is helpful, my script is:

```
import aplpy
import matplotlib.pyplot as plt

f1 = aplpy.FITSFigure("image.fits")
f1.show_colorscale(stretch='linear', vmin=-0.1, vmax=2.0, cmap='gray')
f1.show_regions('catalogue.reg')
plt.show()
plt.close()
```

The catalogue.reg file is a DS9 region file with a list of coordinates in the format:

```
fk5
point(1:08:17.196,-16:04:24.384)  # point=circle color=red dash=1
point(1:20:27.132,-15:20:16.908)  # point=circle color=green dash=1
point(1:16:51.42,-20:52:06.6)  # point=circle color=green dash=1
```

The image is a simple two-dimensional ZEA projected sky image (I've tried with TAN and SIN projected images also but to no avail).

**Comment by epaell** [comment]
I just reverted back to pyregion 1.2_0 and it started working again. So it appears that something incompatible has changed in pyregion 2.0_0.

**Comment by hsdalgleish** [comment]
This might be irrelevant, but I also had errors with WCS. 
I had to change the shape of my data of the fits file to be 2-D. I also had to set NAXIS = 2 and remove any NAXIS3 and NAXIS4 related values from the header in order to get it to work.

**Comment by boada** [comment]
I am also having this problem. 

> In [3]: gc.show_regions('ds9.reg', color='w')
WARNING: FITSFixedWarning: 'datfix' made the change 'Changed '' to '2013-07-28T12:03:41.213''. [astropy.wcs.wcs]

> WCS Keywords
> ---------------------------------------------------------------------------
> AttributeError                            Traceback (most recent call last)
> <ipython-input-3-011830e468a5> in <module>()
> ----> 1 gc.show_regions('ds9.reg', color='w')
> 
> ~/.local/lib/python3.5/site-packages/aplpy/regions.py in show_regions(self, region_file, layer, **kwargs)
> 
> ~/.local/lib/python3.5/site-packages/aplpy/decorators.py in _auto_refresh(f, *args, **kwargs)
>      23     mydata.nesting = getattr(mydata, 'nesting', 0) + 1
>      24     try:
> ---> 25         return f(*args, **kwargs)
>      26     finally:
>      27         mydata.nesting -= 1
> 
> ~/.local/lib/python3.5/site-packages/aplpy/regions.py in show_regions(self, region_file, layer, **kwargs)
>      63         """
>      64 
> ---> 65         PC, TC = ds9(region_file, wcs.WCS(self._header).sub([wcs.WCSSUB_CELESTIAL]), **kwargs)
>      66 
>      67         # ffpc = self._ax1.add_collection(PC)
> 
> ~/.local/lib/python3.5/site-packages/aplpy/regions.py in ds9(region_file, header, zorder, **kwargs)
>     101 
>     102     # convert coordinates to image coordinates
> --> 103     rrim = rr.as_imagecoord(header)
>     104 
>     105     # pyregion and aplpy both correct for the FITS standard origin=1,1
> 
> ~/.local/lib/python3.5/site-packages/pyregion/core.py in as_imagecoord(self, header)
>      70         r = RegionParser.sky_to_image(zip(self, comment_list),
>      71                                       header)
> ---> 72         shape_list, comment_list = zip(*list(r))
>      73         return ShapeList(shape_list, comment_list=comment_list)
>      74 
> 
> ~/.local/lib/python3.5/site-packages/pyregion/ds9_region_parser.py in sky_to_image(shape_list, header)
>     185                     (shape.coord_format not in image_like_coordformats):
>     186 
> --> 187                 new_coords = convert_to_imagecoord(shape, header)
>     188 
>     189                 l1n = copy.copy(shape)
> 
> ~/.local/lib/python3.5/site-packages/pyregion/wcs_converter.py in convert_to_imagecoord(shape, header)
>      79     coord_list_iter = iter(zip(shape.coord_list, arg_types))
>      80 
> ---> 81     new_wcs = WCS(header)
>      82     pixel_scales = proj_plane_pixel_scales(new_wcs)
>      83 

> ~/.local/lib/python3.5/site-packages/astropy/wcs/wcs.py in __init__(self, header, fobj, key, minerr, relax, naxis, keysel, colsel, fix, translate_units, _do_set)
>     393                     orig_header = header
>     394                     header = fits.Header()
> --> 395                     for dict_key in orig_header.keys():
>     396                         header[dict_key] = orig_header[dict_key]
>     397                 except TypeError:
> 
> AttributeError: 'WCS' object has no attribute 'keys'

I fixed it by changing line 65 of `regions.py`. It seems that line converts the header into a WCS object and then passes it down the line. The rest of the code expects a header object and not the WCS object. I removed all of the WCS  conversion `wcs.WCS(self._header).sub([wcs.WCSSUB_CELESTIAL])' to just `self._header' and things work. 

However, I cannot pass a color to the region parser. 

> In [9]: gc.show_regions('ds9.reg', color='w')
> WARNING: FITSFixedWarning: 'datfix' made the change 'Changed '' to '2013-07-28T12:03:41.213''. [astropy.wcs.wcs]
> ---------------------------------------------------------------------------
> TypeError                                 Traceback (most recent call last)
> <ipython-input-9-011830e468a5> in <module>()
> ----> 1 gc.show_regions('ds9.reg', color='w')
> 
> ~/.local/lib/python3.5/site-packages/aplpy/regions.py in show_regions(self, region_file, layer, **kwargs)
> 
> ~/.local/lib/python3.5/site-packages/aplpy/decorators.py in _auto_refresh(f, *args, **kwargs)
>      23     mydata.nesting = getattr(mydata, 'nesting', 0) + 1
>      24     try:
> ---> 25         return f(*args, **kwargs)
>      26     finally:
>      27         mydata.nesting -= 1
> 
> ~/.local/lib/python3.5/site-packages/aplpy/regions.py in show_regions(self, region_file, layer, **kwargs)
>      65         #PC, TC = ds9(region_file, wcs.WCS(self._header).sub([wcs.WCSSUB_CELESTIAL]), **kwargs)
>      66 
> ---> 67         PC, TC = ds9(region_file, self._header, **kwargs)
>      68 
>      69         # ffpc = self._ax1.add_collection(PC)
> 
> ~/.local/lib/python3.5/site-packages/aplpy/regions.py in ds9(region_file, header, zorder, **kwargs)
>     131     pp, aa = rrim.get_mpl_patches_texts(text_offset=text_offset)
>     132 
> --> 133     PC = ArtistCollection(pp, **kwargs)  # preserves line style (dashed)
>     134     TC = ArtistCollection(aa, **kwargs)
>     135     PC.set_zorder(zorder)
> 
> TypeError: __init__() got an unexpected keyword argument 'color'
> 

You can see the change I made in the traceback.

**Comment by boada** [comment]
I can confirm what @epaell said about going back to `pyregion==1.2.0` seems to fix the issue without modifying any code. It still doesn't let me specify the colors though. I'll also mention it in https://github.com/astropy/pyregion/issues/94.

**Comment by astrofrog** [comment]
@epaell @hsdalgleish @boada - sorry for the long delay in replying, this should finally be fixed in APLpy 2.0. Please try upgrading and feel free to re-open this issue if not!
